Swiftification/Array.swift

```diff
@@ -99,10 +99,10 @@ public extension Array {
     }
     
     /// Sections the ordered array by the string specified in the closure, preserving the order of the original array.
-    func sectionBy(@noescape sectionFunction: (Element) -> String) -> [(title: String, items: [Element])] {
-        var lastTitle = ""
+    func sectionBy(@noescape sectionFunction: (Element) -> String?) -> [(title: String?, items: [Element])] {
```

Comment:
Can you just change this to allow any element? Like:

    func sectionBy<T>(@noescape sectionFunction: (Element) -> T) -> [(T, [Element])]